external-crates/move/crates/move-compiler/src/cfgir/optimize/mod.rs

```diff
@@ -64,12 +72,34 @@ pub fn optimize(
             debug_assert_eq!(count, opt_count);
             break;
         }
-
         // reset the count if something has changed
         if optimization(signature, locals, constants, cfg) {
             count = 0
         } else {
             count += 1
         }
     }
+    // If we aren't optimizing a constant, attempt to coalesce locals and re-optimize.
+    if !is_constant && optimize_2024 {
```

Comment:
Why can we not run local coalescing on constants? Or is it just because it is useless?

Reply:
It doesn't do anything of use, and in some cases will make it so that they cannot be completely reduced to a single value.